Components/Queues/AsyncQueueConsumer.php

```diff
@@ -4,90 +4,156 @@
-    use LoggerAwareTrait;
     use UsesExceptionHandlerTrait;
 
     /**
-     * @var QueueManager
+     * Consumer identifier name.
      */
-    private $manager;
+    const CONSUMER_TAG = 'amqp-consumer-%s-%s';
 
     /**
-     * @var string
+     * @var AsyncQueueDriverInterface
      */
-    private $format;
+    private $driver;
 
     /**
-     * AsyncQueueConsumer constructor.
+     * Set the driver to this class with the properties fulfilled.
      *
-     * @param QueueManager $manager
-     * @param string       $format
+     * @param AsyncQueueDriverInterface $driver
      */
-    public function __construct(QueueManager $manager, string $format = 'json')
+    public function setDriver(AsyncQueueDriverInterface $driver)
     {
-        parent::__construct();
-        $this->manager = $manager;
-        $this->format = $format;
+        $this->driver = $driver;
     }
 
     /**
-     * Consumes messages from the given $endpoint until either the expirationCount reaches 0 or ::stop() is called.
-     *
-     * @param EndpointInterface $endpoint
-     *
-     * @throws \Exception
-     *
-     * @return bool
+     * {@inheritdoc}
      */
-    public function consume(EndpointInterface $endpoint)
+    protected function initialize(EndpointInterface $endpoint)
     {
-        $handler = new AmqpQueueHandler($endpoint, (int) $this->expirationCount, $this->format, $this->serializer);
-        $handler->setExceptionHandler($this->getExceptionHandler());
-
-        if ($this->smartesbHelper) {
-            $handler->setSmartesbHelper($this->smartesbHelper);
-        }
-        if ($this->logger) {
-            $handler->setLogger($this->logger);
-        }
-
-        try {
-            $this->manager->getQueue($this->getQueueName($endpoint))->consume($handler);
-        } finally {
-            $this->manager->disconnect();
+        if (!$this->driver->isConnected()) {
+            $this->driver->connect();
         }
-
-        return true;
     }
 
     /**
+     * Returns the consumer name.
+     *
      * {@inheritdoc}
      */
     public function getName()
     {
-        return 'async_queue_consumer';
+        return sprintf(self::CONSUMER_TAG, gethostname(), getmypid());
     }
 
+    /**
+     * Returns the queue name properly treated with queue prefix.
+     *
+     * @param EndpointInterface $endpoint
+     * @return string
+     */
     protected function getQueueName(EndpointInterface $endpoint): string
     {
         $options = $endpoint->getOptions();
 
         return "{$options[QueueProtocol::OPTION_PREFIX]}{$options[QueueProtocol::OPTION_QUEUE_NAME]}";
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function cleanUp(EndpointInterface $endpoint)
+    {
+        $this->driver->disconnect();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function confirmMessage(EndpointInterface $endpoint, MessageInterface $message)
+    {
+        $this->driver->ack($message->getMessageId());
```

Comment:
MessageInterface doesn't guarantee that the message object will have the method `getMessageId()` because you defined this in the QueueMessageInterface and not in the MessageInterface

Reply:
Alright, I changed the type hint to QueueMessageInterface.